liquibase/liquibase
# CLI no longer reading defaultsFile from classpath

Fixes #1892

- - -
## Dev Handoff Notes (Internal Use)
#### Links
* Fixed Issue: liquibase/liquibase#1892
* Local Branch: [https://github.com/liquibase/liquibase/tree/LB-1795](https://github.com/liquibase/liquibase/tree/LB-1795)
* Unit Tests: [https://github.com/liquibase/liquibase/pull/1893/checks](https://github.com/liquibase/liquibase/pull/1893/checks)
* Integration Tests: (shown in Unit Tests link above)
* Functional Tests: [https://jenkins.datical.net/job/liquibase-pro/job/LB-1795/](https://jenkins.datical.net/job/liquibase-pro/job/LB-1795/)

#### Testing
* Steps to Reproduce: liquibase/liquibase#1892
* Guidance:
  * Change only impacts CLI

#### Dev Verification
(did not save output at the time)

## Test Requirements (Liquibase Internal QA)
**Manual Tests**

:: SETUP :: 

This test requires a non-executable Jar containing the liquibase.properties and changelog file. We will create this Jar and add it to the Java classpath when we run Liquibase. Use the attached changelog during the setup steps and also when running the tests. 

* Put your properties file, database driver and the the liquibase/lib directory in one directory.
  * This makes the command line easier to write/read without having a bunch of long paths.
  * The attached changelog is fairly generic, although the createView may fail to deploy if you are not running on Postgres.
* Create the Liquibase resources Jar
  * Include only the changelog and the properties file.
  * `jar cvf liquibase-stuff.jar liquibase.properties postgres_community_changelog.xml`
* Put the liquibase.jar in the same directory as the liquibase-stuff.jar

_Verify liquibase update reads properties and changelog within liquibase-stuff.jar._

* `java -cp "liquibase.jar;lib/*;liquibase-stuff.jar;postgresql-42.2.5.jar" liquibase.integration.commandline.Main --logLevel=info update`
* Successful output from  3.5.3 is attached; it should look similar in the fix build.

**Automated Tests**

No new functional tests required for this fix; the PR submitter created an integration test: testParseDefaultPropertyFilesFromResources. I love seeing tests!!



┆Issue is synchronized with this [Jira Story](https://datical.atlassian.net/browse/LB-1796) by [Unito](https://www.unito.io)
